include/golioth/ota.h

```diff
@@ -68,6 +68,13 @@ struct golioth_ota_component
     int32_t size;
     /// True, if the component is compressed and requires decompression
     bool is_compressed;
+    /// Artifact Hash
+    char hash[CONFIG_GOLIOTH_OTA_MAX_VERSION_LEN + 1];
```

Comment:
Should these constants be different? I'm also wondering if these should be dynamically allocated.

Reply:
They should, the hash is limited to 64 characters, I propose to leave it as a constant. Bootloader is either "mcuboot" or "default", so 7 characters. We can also discuss removing it from the payload (@hasheddan).
It makes the most sense to dynamically allocate the memory for the URI.